test/services/data_model_engine.js

```diff
@@ -515,6 +503,85 @@ describe('Data Model Engine', () => {
       expect(ret.results).to.equal(assetSet);
       expect(ret.resultCount).to.equal(165);
     });
+
+    describe('Finding assets by identifiers', () => {
+      let mockFindEventQueryObject;
+      let mockFindEventQueryObjectFactory;
+      let mockAccountAccessDefinitions;
+      let result;
+      const identifier = {
+        isbn: 'abc-def',
+        gs1: '123'
+      };
+      const token = {one: 1};
+      const accessLevel = 2;
+
+      before(async () => {
+        mockFindEventQueryObjectFactory = {
+          create: sinon.stub()
+        };
+
+        mockFindEventQueryObject = {
+          execute: sinon.stub()
+        };
+
+        mockAccountAccessDefinitions = {
+          getTokenCreatorAccessLevel: sinon.stub()
+        };
+
+        mockEntityBuilder.validateAndCastFindAssetsParams.returns({identifier});
+        mockFindEventQueryObjectFactory.create.returns(mockFindEventQueryObject);
+        mockFindEventQueryObject.execute.resolves({
+          results: [
+            {content: {idData: {assetId: '1', foo: 'bar'}}},
+            {content: {idData: {assetId: '10', one: {two: 2}}}},
+            {content: {idData: {assetId: '1'}}}
+          ]
+        });
+        mockAccountAccessDefinitions.getTokenCreatorAccessLevel.resolves(accessLevel);
+
+        modelEngine = new DataModelEngine({}, {}, mockEntityBuilder, {},
+          {}, {}, {}, mockFindEventQueryObjectFactory, {}, mockFindAssetQueryObjectFactory, mockAccountAccessDefinitions);
+
+        result = await modelEngine.selectAssetsByIdentifier(identifier, accessLevel);
+      });
+
+      it('should create FindEventQueryObject with correct parameters', () => {
+        expect(mockFindEventQueryObjectFactory.create).to.be.calledWith({
+          data: {
+            type: 'ambrosus.event.identifiers',
+            isbn: 'abc-def',
+            gs1: '123'
+          }
+        }, accessLevel);
+      });
+
+      it('should execute and return unique ids', () => {
+        expect(mockFindEventQueryObject.execute).to.be.calledOnce;
+        expect(result).to.deep.equal(['1', '10']);
+      });
+
+      it('when accessLevel was not given it defaults to 0', async () => {
+        await modelEngine.selectAssetsByIdentifier(identifier);
```

Comment:
Tests for different subjects should be grouped at  least into separate describe blocks